src/storage/cloud_storage.rs

```diff
@@ -73,6 +73,23 @@ fn item_to_metadata(item: Item) -> ObjectMetadata {
     }
 }
 
+fn item_to_file_info(item: Item) -> Fileinfo<PathBuf, ObjectMetadata> {
+    Fileinfo {
+        path: PathBuf::from(item.name),
+        metadata: ObjectMetadata {
+            last_updated: match u64::try_from(item.updated.timestamp_millis()) {
+                Ok(timestamp) => SystemTime::UNIX_EPOCH.checked_add(Duration::from_millis(timestamp)),
+                _ => None,
+            },
+            is_file: true,
+            size: match item.size.parse() {
+                Ok(size) => size,
+                //TODO: return 450
+                _ => 0,
+            },
```

Comment:
In this case, I think you can do `item.size.parse().or_else(0)`, rather than `match`.

Reply:
The TODO will solve this